### DISCUSSION TITLE
casadi.Function out of several casadi.Function / combining functions

### DISCUSSION THREAD

**Discussion by Sofie967** [casadi/casadi#3233] [Q&A] [answered]
Hello,

I am new to casadi and I was wondering if I could combine several casadi.Function objects to one function.

So something like this (using casadi in in MATLAB):

`x = MX.sym('x');`
`y1 = Function('y1',{x},{x^2});`
`y2 = Function('y1',{x},{x});`
`y = y1 + y2;                        % y is casadi.Function object  y = x^2 + x`

With this syntax it definitely does not work, but maybe there is a way for combining functions?

I hope my question is understandable. Thank you for any help!

**Comment by jgillis** [comment, marked as answer]
Arithmatic operations should be performed on expressions, not on Functions. The trick is to call Functions symbolically to obtain expressions:
```
y = Function('y',{x},{y1(x)+y2(x)})
```

**Reply by Sofie967** [reply]
Perfect, this works. And makes sense, I could have thought of that. Thank you very much!
